---
title: "centralperk"
output: rmarkdown::pdf_document
vignette: >
  %\VignetteIndexEntry{centralperk}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(centralperk)
```

## Group28 GitHub repository:

<https://github.com/rforbiodatascience24/Group_28_package>

## Description of this package

The package centralperk contains five functions which represent the core processes of the central dogma in molecular biology. They can be used alone or in collaboration, in order to demonstrate different steps of the central dogma.

### Generate nucleotide sequence

Generates a nucleotide sequence based on input defining the length. This resembles the DNA of an organism.

```{r}
dna_sequence <- centralperk::sample_nucleotides(99)

dna_sequence
```

### DNA to RNA

Converts the input DNA to RNA. This resembles the transcription of the DNA to mRNA in the organism.

```{r}
rna_sequence <- centralperk::dna_to_rna(dna_sequence)

rna_sequence
```

### RNA sequence to codon list

This function recognises the codons making up the RNA sequence. This resembles the tRNA recognises the mRNA in the ribosome.

```{r}
codons <- centralperk::create_codons(rna_sequence,1)

codons
```

### Codons to amino acids

This function looks up the codons in the 'codon_table' and the corresponding amino acid one-letter code gets appended, generating an amino acid sequence from the list of codons. This resembles the ribosome generating the peptide.

```{r}
AA_sequence <- centralperk::translate(codons)

AA_sequence
```

### Plot of amino acids

Plots the amino acid distributions of a string.

```{r}
p1 <- centralperk::plot_codons(AA_sequence)

p1
```

## Discussion

The package can be useful for simulation of the central dogma. The different functions give the possibility of many different outputs. For example, if the DNA bases are known, and the protein sequence is wanted, or if the RNA bases are known and a plot of the number of each amino acids is wanted.
The package includes 2 dependencies. The number of dependencies is limited to this, as using too many dependencies reduces the reproductability of the function.

In addition to the functions included in the package, another function which could be included is a reverse transcription function, which could transcribe from RNA to DNA bases.
